src/geom/Endpoint.cc

```diff
@@ -136,17 +124,26 @@ ndarray::Array<double, 2, 2> GenericEndpoint::arrayFromData(ndarray::Array<doubl
     return ndarray::copy(data);
 }
 
-template <int N>
-PointEndpoint<N>::PointEndpoint(int nAxes) : BaseVectorEndpoint<Point<double, N>>(N) {
-    if (nAxes != N) {
+Point2Endpoint::Point2Endpoint(int nAxes) : BaseVectorEndpoint<Point2D>(2) {
+    if (nAxes != 2) {
         std::ostringstream os;
-        os << "nAxes = " << nAxes << " != " << N;
+        os << "nAxes = " << nAxes << " != 2";
         throw LSST_EXCEPT(pexExcept::InvalidParameterError, os.str());
     }
 }
 
-template <int N>
-void PointEndpoint<N>::normalizeFrame(std::shared_ptr<ast::Frame> framePtr) const {
+std::vector<Point2D> Point2Endpoint::arrayFromData(ndarray::Array<double, 2, 2> const& data) const {
+    this->_assertNAxes(this->_getNAxes(data));
+    int const nPoints = this->_getNPoints(data);
+    Array array;
+    array.reserve(nPoints);
+    for (auto const& dataCol : data.transpose()) {
+        array.emplace_back(dataCol[0], dataCol[1]);
```

Comment:
@r-owen Isn't `dataCol` a 1×2 `ndarray::Array`? What's to stop you from declaring an array of length N, filling it from `dataCol`, and passing that in?